core/services/workflows/syncer/handler.go

```diff
@@ -499,6 +486,61 @@ func (h *eventHandler) workflowRegisteredEvent(
 	return nil
 }
 
+// getWorkflowArtifacts retrieves the workflow artifacts from the database if they exist,
+// or downloads them from the specified URLs if they are not found in the database.
+func (h *eventHandler) getWorkflowArtifacts(
+	ctx context.Context,
+	payload WorkflowRegistryWorkflowRegisteredV1,
+) ([]byte, []byte, error) {
+	spec, err := h.orm.GetWorkflowSpec(ctx, hex.EncodeToString(payload.WorkflowOwner), payload.WorkflowName)
```

Comment:
@agparadiso It's important to fetch by the workflow ID and NOT by the owner + the name; this will guarantee that the contents of the artifacts are the same as what's in the registration event.